app/assets/scripts/components/documents/single-edit/index.js

```diff
@@ -249,3 +268,33 @@ const SaveButton = () => {
     </Tip>
   );
 };
+
+const SaveAndContinueButton = ({ nextStep }) => {
+  const history = useHistory();
+  const { dirty, isSubmitting, submitForm, status } = useFormikContext();
+  const submitAndContinue = useCallback(
+    (e) => {
+      submitForm(e).then(({ error, data }) => {
+        if (!error && nextStep) {
+          history.replace(documentEdit(data.alias, data.version, nextStep));
+        }
+      });
+    },
+    [submitForm, history, nextStep]
+  );
```

Comment:
Can we use `async/await` here to be consistent with the rest of the app?

Reply:
👍